### DISCUSSION TITLE
Jukebox mode block list

### DISCUSSION THREAD

**Discussion by john-** [jcorporation/myMPD#1057] [Ideas]
I can see the value in having a "block list" for jukebox mode.  Basically, the user would define a list of songs or albums that they do not want to have played when in jukebox mode.

Some examples of things I want to block:

- Christmas songs throughout most of the year
- Songs that I have to be in a very specific mood to listen to
- Songs that I probably should just delete but can't bring myself to do it

Maybe playlists can be defined to include things to block.  In jukebox configuration the user can select one or more playlists to use as a filter when picking random songs to play.

**Comment by jcorporation** [comment]
Why do you not create a playlist from which the jukebox should select songs? You can create a smart playlist from a search that excludes e.g. genres.

**Reply by john-** [reply]
I have over 35K songs. Most of these I want to come up randomly.  Therefore my first thought was to reduce from that number vs. build a list from scratch.  However, I will take a look at a smart playlist that uses a well defined search.

I gave up on trying to determining genres years ago.  Back then each song could have only one genre ("jazz" or "christmas").  Even with genre there are just some songs within a genre I really don't want to come up randomly.

**Reply by jcorporation** [reply]
MPD and myMPD supports multiple genres per song.

**Comment by john-** [comment]
Sounds good.  I can tag just the songs I don't want to listen to :)

**Comment by hjri** [comment]
There seems to be no way to limit jukebox in album mode - it picks albums from entire database, no option to limit it to a playlist or like.

Result is that it often picks "albums" which are just separate tunes from C64 collection since every tune is also an "album" because it may have subtunes.

**Reply by jcorporation** [reply]
Yes, this is not implemented. Picking only songs from a playlist was easier to implement than picking albums from a playlist and I wanted to implement it later, but I have forgotten to add it to my todo list.

**Reply by hjri** [reply]
naturally. on top of that using traditional playlists for albums is quite inconvenient, since you have to add tons of songs to them, possibly overloading (my)MPD if you try to add an entire directory or several albums at once.

It would be instead much nicer to be able to set search criteria for album jukebox with ability to set negative filters. Being able to exclude filesystem paths i.e. `Soundtracks/*` or `*.sid` would be a godsend. You could so something like use already existing search functionality to build the list of albums to use, and then when picking random album, check if any files match the filter and if they do(n't) - drop entire album and try again.

**Reply by jcorporation** [reply]
I do not like negative lists based on filesystem paths. We should go with tags.

One idea could be:
- Implement smart album playlists that are based on album search filters
- Let the jukebox select from such a playlist.

**Reply by hjri** [reply]
i don't like anything filesystem-based but it is convenient when tags are insufficient.

**Comment by jcorporation** [comment]
myMPD v13 (not yet released) adds a include and a exclude filter. Both works for jukebox album and song mode. The filters are native mpd filter expressions.
